File: php/test/Behavioral/ChainOfResponsibilityTest.php



<?

#//// Original copy of this content taken from http://www.fluffycat.com/Java-Design-Patterns/ in 2010
#//// Original Author: Larry Truett
#//// Privacy Policy at http://www.fluffycat.com/Privacy-Policy/
#//Chain of Responsibility Overview
#//A method called in one class will move up a class hierarchy until a method is found that can properly handle the call.

require_once 'src/Behavioral/ChainOfResponsibility/TopTitle.php';
require_once 'src/Behavioral/ChainOfResponsibility/DvdCategory.php';
require_once 'src/Behavioral/ChainOfResponsibility/DvdSubCategory.php';
require_once 'src/Behavioral/ChainOfResponsibility/DvdSubSubCategory.php';

class Behavioral_ChainOfResponsibilityTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {

		$topTitle;
		$allCategories;

		$comedy = new DvdCategory("Comedy");
		$comedy->setTopCategoryTitle("Ghost World");

		$comedyChildrens = new DvdSubCategory($comedy, "Childrens");

		$comedyChildrensAquatic = new DvdSubSubCategory($comedyChildrens, "Aquatic");
		$comedyChildrensAquatic->setTopSubSubCategoryTitle( "Sponge Bob Squarepants");

		$topTitle = $comedy->getTopTitle();
		$allCategories = $comedy->getAllCategories();
		$this->assertEquals($topTitle,'Ghost World','DvdCateogry getTopTitle test');
		$this->assertEquals($allCategories,'Comedy','DvdCategory getAllCategories test');

		$topTitle = $comedyChildrens->getTopTitle();
		$allCategories = $comedyChildrens->getAllCategories();
		$this->assertEquals($topTitle,'Ghost World','DvdSubCateogry getTopTitle test');
		$this->assertEquals($allCategories,'Comedy/Childrens','DvdSubCategory getAllCategories test');

		$topTitle = $comedyChildrensAquatic->getTopTitle();
		$allCategories = $comedyChildrensAquatic->getAllCategories();
		$this->assertEquals($topTitle,'Sponge Bob Squarepants','DvdSubSubCateogry getTopTitle test');
		$this->assertEquals($allCategories,'Comedy/Childrens/Aquatic','DvdSubSubCategory getAllCategories test');
	}
}

File: go/behavioral/chainofresponsibility/chain_of_responsibility_test.go
package chainofresponsibility

import (
	"fmt"
   "testing"
   "../../lib/asserts"
)

var comedy *DvdCategory = NewDvdCategory("Comedy")
var comedyChildrens = NewDvdSubCategory(comedy, "Childrens")
var comedyChildrensAquatic = NewDvdSubSubCategory(comedyChildrens, "Aquatic")


func TestDvdCategory( t *testing.T) {
	comedy.SetTopCategoryTitle("Ghost World")
	var topTitle1 = comedy.GetTopTitle()
	asserts.Equals( t, 
		"DvdCategory test", 
		fmt.Sprintf("The top title for %s is %s" , comedy.GetAllCategories() , topTitle1 ), 
		"The top title for Comedy is Ghost World")
}

func TestDvdSubCategory( t *testing.T) {
	var topTitle2 = comedyChildrens.GetTopTitle()
	asserts.Equals( t, "DvdSubCategory test", 
		fmt.Sprintf("The top title for %s is %s" , comedyChildrens.GetAllCategories() , topTitle2), 
		"The top title for Comedy/Childrens is Ghost World")
}

func TestDvdSubSubCategory( t *testing.T) {
	comedyChildrensAquatic.SetTopSubSubCategoryTitle( "Sponge Bob Squarepants")
	var topTitle3 = comedyChildrensAquatic.GetTopTitle()
	asserts.Equals( t, "DvdSubSubCategory test", 
		fmt.Sprintf("The top title for %s is %s" , comedyChildrensAquatic.GetAllCategories() , topTitle3), 
		"The top title for Comedy/Childrens/Aquatic is Sponge Bob Squarepants" )
}
